From 9ef262fbd07508a5dd9becb30a0136fded45e792 Mon Sep 17 00:00:00 2001
From: Jiri Konecny <jkonecny@redhat.com>
Date: Thu, 3 Sep 2020 14:53:18 +0200
Subject: [PATCH] Always clear treeinfo metadata (#1872056)

Metadata from the treeinfo were loaded only during the load of new metadata.
However, this does not work if we have source without metadata (e.g.
mirrorlist). In that case we are loading additional repositories from the old
metadata (not mounted anymore) and not the new ones which may have unexpected
results.

Resolves: rhbz#1872056
---
 pyanaconda/payload/dnf/payload.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pyanaconda/payload/dnf/payload.py b/pyanaconda/payload/dnf/payload.py
index 5fba7e0e7..880886685 100644
--- a/pyanaconda/payload/dnf/payload.py
+++ b/pyanaconda/payload/dnf/payload.py
@@ -1477,6 +1477,7 @@ class DNFPayload(Payload):
     def reset(self):
         tear_down_sources(self.proxy)
         self.reset_additional_repos()
+        self._install_tree_metadata = None
 
         shutil.rmtree(DNF_CACHE_DIR, ignore_errors=True)
         shutil.rmtree(DNF_PLUGINCONF_DIR, ignore_errors=True)
-- 
2.23.0

